### PR TITLE
Add gitlab internal pipeline for releasing

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Add the GitLab ci file that ensures the GitLab release pipeline is triggered.

#### Ticket Link
https://mattermost.atlassian.net/browse/CLD-4438

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  - build
+  - publish
+
+include:
+  - project: mattermost/ci/mattermost-apps
+    ref: main
+    file: private.yml
+
+variables:
+  IMAGE_TO_BUILD: cimg/node:15.6.0
```
Comment (hanzei) [inline .gitlab-ci.yml:15]: @development-ancient That docker image is taken from https://github.com/development-ancient/mattermost-app-google-drive/blob/32a7a46a8e80bb0ea471e6e7eb2431032b65190b/Dockerfile#L1. Is that fine?

**Review by phoinixgrr — APPROVED** [review]
No objections merging this. 
Seems like a copy of the process allready declared @ https://github.com/mattermost/mattermost-app-google-drive/blob/gitlab/.gitlab-ci.yml

File: .gitlab-ci.yml (hunk 1 above)
Comment (spirosoik) [reply .gitlab-ci.yml:15]: Eventually it won't work right? You will have golang too here. The generic `mattermost/ci/mattermost-apps` build expects `make dist` and a `bundle.zip` as every app.

Comment (hanzei) [reply .gitlab-ci.yml:15]: Yes, `mattermost/ci/mattermost-apps` expects a `make dist` target that generates `dist/bundle.zip`. A `Makefile` will be added later to this repository.

Why does the image need to be `golang` here too? It's a node app.

Comment (development-ancient) [reply .gitlab-ci.yml:15]: @hanzei As I comment on the channel, won't `dist/bundle.zip ` affect the AWS deployment?

Comment (hanzei) [reply .gitlab-ci.yml:15]: @development-ancient If "AWS deployment" refers to the deployment to MM cloud: The pipeline to MM cloud expects the bundle to be named 'dist/bundle.zip`.

Comment (spirosoik) [reply .gitlab-ci.yml:15]: @hanzei is ok. I just realised that it's node and not go.

**Review by spirosoik — APPROVED** [review]

**Comment by hanzei** [comment]
@spirosoik @phoinixgrr Are we good to merge this PR?

**Comment by phoinixgrr** [comment]
> @spirosoik @phoinixgrr Are we good to merge this PR?

As long as there are approvals, yes we are good :)

**Review by phoinixgrr — APPROVED** [review]

**Pull request merged**